docs/diagrams/terminateThenReplaceSchedulerStateTransition.puml

```diff
@@ -0,0 +1,16 @@
+@startuml
+[*] --> state2
```

Comment:
From the sequence table, the state3 should be first one to start?  I'm not sure if this diagram better than the table list, it makes me really hard to understand like why `state2 -> state1` and `state5 -> state1` and `state4 -> state1`. It kinds of mixing the states and two tasks.

Reply:
The table lists the sequence of how an example deployment works, while the diagram describes the steady state. So state1 ~ state5 doesn't have any sequential order. Maybe renaming them into stateA~E would be more helpful? 

I think what's missing from appendix A is what would be like in a steady state. Also, we should separate the "condition" (the cluster snapshot that we can read) and the "action" (the action that the scheduler take) in the sequence table. that's why I add one more column in the sequence table here.

Reply:
Hm. Perhaps giving the states meaningful names will make this clearer? E.g. state2 is "Empty", state1 is "Active", state3 is "OutdatedRunning", state4 is "OutdatedStopped", state5 is "Failing".

Reply:
Let me change the name to the actual behavior of these states!